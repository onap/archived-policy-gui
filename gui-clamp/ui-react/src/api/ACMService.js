/*
 * ============LICENSE_START=======================================================
 *  Copyright (C) 2021-2022 Nordix Foundation.
 *  ================================================================================
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 *  SPDX-License-Identifier: Apache-2.0
 *  ============LICENSE_END=========================================================
 */

export default class ACMService {

  static async getACMInstantiation() {

    const response = await fetch(window.location.pathname +
      'onap/policy/clamp/acm/v2/instantiation');

    return response
  }

  static async createInstanceProperties(instancePropertiesTemplate) {

    const response = await fetch(window.location.pathname +
      'onap/policy/clamp/acm/v2/instanceProperties', {
      method: 'POST',
      headers: {
        "Content-Type": "application/json"
      },
      credentials: 'same-origin',
      body: JSON.stringify(instancePropertiesTemplate),
    });

    return response
  }

  static async updateInstanceProperties(instanceName, version, instancePropertiesTemplate) {
    const params = {
      name: instanceName,
      version: version
    }

    const response = await fetch(window.location.pathname +
        'onap/policy/clamp/acm/v2/instanceProperties?' + (new URLSearchParams(params)), {
      method: 'PUT',
      headers: {
        "Content-Type": "application/json"
      },
      credentials: 'same-origin',
      body: JSON.stringify(instancePropertiesTemplate),
    });

    return response
  }

  static async deleteInstantiation(name, version) {
    const params = {
      name: name,
      version: version
    }

    const response = await fetch(window.location.pathname +
      'onap/policy/clamp/acm/v2/instanceProperties?' + (new URLSearchParams(params)), {
      method: 'DELETE',
      credentials: 'same-origin',
    });

    const data = await response;

    return data;
  }

  static async getInstanceOrderState(name, version) {
    const params = {
      name: name,
      version: version
    }
    const response = await fetch(window.location.pathname +
      'onap/policy/clamp/acm/v2/instantiationState'+ '?' + (new URLSearchParams(params)));

    const data = await response;

    return data;
  }

  static async changeInstanceOrderState(toscaObject) {
    const response = await fetch(window.location.pathname +
      'onap/policy/clamp/acm/v2/instantiation/command', {
      method: 'PUT',
      headers: {
        "Content-Type": "application/json"
      },
      credentials: 'same-origin',
      body: JSON.stringify(toscaObject)
    });

    return response
  }

  static async getToscaTemplate(name, version, instanceName) {
    const params = instanceName != null ?
      {
        name: name,
        version: version,
        instanceName: instanceName
      } :
      {
        name: name,
        version: version
      }

    const response = await fetch(window.location.pathname +
      'onap/policy/clamp/acm/v2/commission/toscaservicetemplate' + '?' + (new URLSearchParams(params)));

    const data = await response;

    return data;
  }

  static async uploadToscaFile(toscaObject) {
    const response = await fetch(window.location.pathname +
      'onap/policy/clamp/acm/v2/commission', {
      method: 'POST',
      headers: {
        "Content-Type": "application/json"
      },
      credentials: 'same-origin',
      body: JSON.stringify(toscaObject),
    });

    return response

  }

  static async deleteToscaTemplate(name, version) {
    const params = {
      name: name,
      version: version
    }

    const response = await fetch(window.location.pathname +
      'onap/policy/clamp/acm/v2/commission' + '?' + (new URLSearchParams(params)),
      {
        method: 'DELETE'
      });

    const data = await response;

    return data;
  }

  static async getCommonOrInstanceProperties(name, version, instanceName, isCommon) {
    const params = instanceName != null ?
      {
        name: name,
        version: version,
        instanceName: instanceName,
        common: isCommon
      } :
      {
        name: name,
        version: version,
        common: isCommon
      }

    const response = await fetch(window.location.pathname +
      'onap/policy/clamp/acm/v2/commission/getCommonOrInstanceProperties' + '?' + (new URLSearchParams(params)));

    return response;
  }

}
