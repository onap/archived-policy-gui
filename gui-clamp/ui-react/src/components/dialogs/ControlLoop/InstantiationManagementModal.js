/*
 * ============LICENSE_START=======================================================
 *  Copyright (C) 2021 Nordix Foundation.
 *  ================================================================================
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 *  SPDX-License-Identifier: Apache-2.0
 *  ============LICENSE_END=========================================================
 */

import styled from "styled-components";
import Modal from "react-bootstrap/Modal";
import Button from "react-bootstrap/Button";
import React, { useContext, useEffect, useRef, useState } from "react";
import InstantiationOrderStateChangeItem from "./InstantiationOrderStateChangeItem";
import ControlLoopService from "../../../api/ControlLoopService";
import { Alert, Container, Dropdown } from "react-bootstrap";

const ModalStyled = styled(Modal)`
  @media (min-width: 800px) {
    .modal-xl {
      max-width: 96%;
    }
  }
  background-color: transparent;
`

const DivWhiteSpaceStyled = styled.div`
  overflow: auto;
  min-width: 100%;
  max-height: 300px;
  padding: 5px 5px 0px 5px;
  text-align: center;
`

const AlertStyled = styled(Alert)`
  margin-top: 10px;
`

const InstantiationManagementModal = (props) => {
  const [show, setShow] = useState(true);
  const [windowLocationPathnameGet, setWindowLocationPathnameGet] = useState('');
  const [windowLocationPathNameSave, setWindowLocationPathNameSave] = useState('');
  const [controlLoopIdentifierList, setControlLoopIdentifierList] = useState([]);
  const [orderedState, setOrderedState] = useState('');
  const [toscaOrderStateObject, setToscaOrderStateObject] = useState({});
  const [instantiationOrderStateOk, setInstantiationOrderStateOk] = useState(true);
  const [instantiationOrderStateError, setInstantiationOrderStateError] = useState({});
  const [alertMessage, setAlertMessage] = useState(null);

  useEffect(async () => {
    setWindowLocationPathnameGet(window.location.pathname);

    const instantiationOrderState = await ControlLoopService.getInstanceOrderState(windowLocationPathnameGet)
      .catch(error => error.message);

    const orderStateJson = await instantiationOrderState.json();

    if (!instantiationOrderState.ok || orderStateJson['controlLoopIdentifierList'].length === 0) {
      setInstantiationOrderStateOk(true);
      setInstantiationOrderStateError(orderStateJson);
    } else {
      setControlLoopIdentifierList(orderStateJson['controlLoopIdentifierList']);
      setOrderedState(orderStateJson['orderedState']);
    }
  }, []);

  const handleDropSelect = (event) => {
    console.log("handleDropDownChange called");

    const stateChangeObject = {
      orderedState: event,
      controlLoopIdentifierList: controlLoopIdentifierList
    }
    setToscaOrderStateObject(stateChangeObject);
    orderStateContext.orderState = stateChangeObject;
  }

  const handleSave = async () => {
    console.log("handleSave called");
    setWindowLocationPathNameSave(window.location.pathname);

    const response = await ControlLoopService.changeInstanceOrderState(toscaOrderStateObject, windowLocationPathNameSave).catch(error => error.message);

    if (response.ok) {
      successAlert();
    } else {
      await errorAlert(response);
    }
  }

  const handleClose = () => {
    console.log('handleClose called');
    setShow(false);
    props.history.push('/');
  }

  const successAlert = () => {
    console.log("successAlert called");
    setAlertMessage(<Alert variant="success">
      <Alert.Heading>Order State Changed Success</Alert.Heading>
      <p>Order State Changed was successfully changed</p>
      <hr/>
    </Alert>);
  }

  const errorAlert = async (response) => {
    console.log("errorAlert called");
    setAlertMessage(<Alert variant="danger">
      <Alert.Heading>Order State Changed Failure</Alert.Heading>
      <p>An error occurred while trying to change order state</p>
      <p>Status code: { await response.status } : { response.statusText }</p>
      <p>Status Text: { await response.text() }</p>
      <hr/>
    </Alert>);
  }

  return (
    <ModalStyled size="sm"
                 show={ show }
                 onHide={ handleClose }
                 backdrop="static"
                 keyboard={ false }>
      <Modal.Header closeButton>
        <Modal.Title>Manage Instantiation</Modal.Title>
      </Modal.Header>
      <div style={ { padding: '5px 5px 0 5px' } }>
        <Modal.Body>
          <Container>
            <Dropdown onSelect={ handleDropSelect }>
              <Dropdown.Toggle variant="dark" id="dropdown-basic">
                Select Order State
              </Dropdown.Toggle>
              <Dropdown.Menu>
                <Dropdown.Item eventKey="UNINITIALISED">UNINITIALISED</Dropdown.Item>
                <Dropdown.Item eventKey="PASSIVE">PASSIVE</Dropdown.Item>
                <Dropdown.Item eventKey="RUNNING">RUNNING</Dropdown.Item>
              </Dropdown.Menu>
            </Dropdown>
            {
              controlLoopIdentifierList.map((clIdList, index) => (
                <InstantiationOrderStateChangeItem title={ clIdList.name } index={ index } key={ index } />
              ))
            }
          </Container>
          <AlertStyled show={ !instantiationOrderStateOk }
                       variant="danger">Can't get instantiation ordered state:<br/>{ JSON.stringify(instantiationOrderStateError, null, 2) }</AlertStyled>
        </Modal.Body>
        <DivWhiteSpaceStyled>
          { alertMessage }
        </DivWhiteSpaceStyled>
      </div>
      <Modal.Footer>
        <Button variant="primary" onClick={ handleSave }>Save</Button>
        <Button variant="secondary" onClick={ handleClose }>Close</Button>
      </Modal.Footer>
    </ModalStyled>
  );
}

export default InstantiationManagementModal;
