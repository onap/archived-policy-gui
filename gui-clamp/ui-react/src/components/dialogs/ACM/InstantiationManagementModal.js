/*
 * ============LICENSE_START=======================================================
 *  Copyright (C) 2022 Nordix Foundation.
 *  ================================================================================
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 *  SPDX-License-Identifier: Apache-2.0
 *  ============LICENSE_END=========================================================
 *
 *
 */

import Modal from "react-bootstrap/Modal";
import { Alert, Container, Table } from "react-bootstrap";
import Button from "react-bootstrap/Button";
import React, { useEffect, useState } from "react";
import styled from "styled-components";
import { Link } from "react-router-dom";
import ACMService from "../../../api/ACMService";
import Row from "react-bootstrap/Row";
import InstantiationUtils from "./utils/InstantiationUtils";

const ModalStyled = styled(Modal)`
  background-color: transparent;
`

const HorizontalSpace = styled.div`
  padding-right: 2px;
  padding-left: 2px;
`;

const DivWhiteSpaceStyled = styled.div`
  overflow: auto;
  min-width: 100%;
  max-height: 300px;
  padding: 5px 5px 0px 5px;
  text-align: center;
`
const InstantiationManagementModal = (props) => {
  const [show, setShow] = useState(true);
  const [instantiationList, setInstantiationList] = useState([]);
  const [alertMessage, setAlertMessage] = useState(null);

  useEffect(async () => {

    const response = await ACMService.getACMInstantiation();

    const instantiationListJson = await response.json();

    const parsedInstantiationList = InstantiationUtils.parseInstantiationList(instantiationListJson['automationCompositionList']);

    setInstantiationList(parsedInstantiationList);
  }, []);

  const getBackgroundColor = (index) => {
    if (index % 2 === 0) {
      return 'Silver';
    }

    return 'White';
  }

  const deleteInstantiationHandler = async (index, instantiation) => {
    console.log("deleteInstantiationHandler called");

    if (instantiation.disableDelete) {
      return;
    }

    const name = instantiation.name;
    const version = instantiation.version;

    const response = await ACMService.deleteInstantiation(name, version);

    updateList(index);

    if (response.ok) {
      successAlert();
    } else {
      await errorAlert(response);
    }
  }

  const updateList = (index) => {
    console.log("updateList called")

    const updatedList = [...instantiationList];
    updatedList.splice(index, 1);

    setInstantiationList(updatedList);
  }

  const handleClose = () => {
    console.log("handleClose called");
    setShow(false);
    props.history.push('/');
  }

  const successAlert = () => {
    console.log("successAlert called");
    setAlertMessage(<Alert variant="success">
      <Alert.Heading>Deletion of Instantiation Success</Alert.Heading>
      <p>Deletion of Instantiation was successful!</p>
      <hr/>
    </Alert>);
  }

  const errorAlert = async (response) => {
    console.log("errorAlert called");
    setAlertMessage(<Alert variant="danger">
      <Alert.Heading>Deletion of Instantiation Failure</Alert.Heading>
      <p>An error occurred while trying to delete instantiation</p>
      <p>Status code: { await response.status } : { response.statusText }</p>
      <p>Status Text: { await response.text() }</p>
      <hr/>
    </Alert>);
  }

  const clearErrors = () => {
    console.log("clearErrors called");
    setAlertMessage(null);
  }

  return (
    <ModalStyled size="xl"
                 show={ show }
                 onHide={ handleClose }
                 backdrop="static"
                 keyboard={ false }>
      <Modal.Header closeButton>
        <Modal.Title>Manage Instances</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <Container>
          <Row>
            <Link to={ { pathname: "/editACMInstanceProperties" } }>
              <Button variant="primary" type="null">Create Instance</Button>
            </Link>
            <HorizontalSpace/>
            <Link to={ { pathname: "/monitorInstantiation" } }>
              <Button variant="secondary" type="null">Monitor Instantiations</Button>
            </Link>
          </Row>
        </Container>
        <Table bordered style={ { marginTop: '10px' } }>
          <thead>
          <tr>
            <th>#</th>
            <th style={ { textAlign: "center" } }>Instantiation Name</th>
            <th style={ { textAlign: "center" } }>Edit Instantiation</th>
            <th style={ { textAlign: "center" } }>Delete Instantiation</th>
            <th style={ { textAlign: "center" } }>Change Order State</th>
            <th style={ { textAlign: "center" } }>Instantiation Order State</th>
            <th style={ { textAlign: "center" } }>Instantiation Current State</th>
          </tr>
          </thead>
          <tbody>
          { instantiationList.map((instantiation, index) => {
            return (
              <tr style={ { backgroundColor: getBackgroundColor(index) } } key={ index } className="instantiationList">
                <td>{ index + 1 }</td>
                <td>{ instantiation.name }</td>
                <td style={ { textAlign: "center" } }>
                  <Link to={ {
                    pathname: "editACMInstanceProperties",
                  } } state={ instantiation.name }>
                    <Button variant="outline-success" type="null"
                            disabled={ true }
                            style={ { cursor: "not-allowed" } }>Edit</Button>
                  </Link>
                </td>
                <td style={ { textAlign: "center" } }>
                  <Button variant={ instantiation.disabled ? "outline-danger" : "danger" } type="null"
                          disabled={ instantiation.disableDelete }
                          style={ instantiation.disableDelete ? { cursor: "not-allowed" } : {} }
                          onClick={(e) => deleteInstantiationHandler(index, instantiation)}>Delete</Button>
                </td>
                <td style={ { textAlign: "center" } }>
                  <Link to={ {
                    pathname: "changeOrderState",
                    instantiationName: instantiation.name,
                    instantiationVersion: instantiation.version
                  } }>
                    <Button variant="secondary" type="null">Change</Button>
                  </Link>
                </td>
                <td>{ instantiation.orderedState }</td>
                <td>{ instantiation.currentState }</td>
              </tr>
            )
          }) }
          </tbody>
        </Table>
        <DivWhiteSpaceStyled>
          { alertMessage }
        </DivWhiteSpaceStyled>
      </Modal.Body>
      <Modal.Footer>
        <Button variant="secondary" type="null" onClick={ clearErrors }>Clear Error Message</Button>
        <Button variant="secondary" type="null" onClick={ handleClose }>Close</Button>
      </Modal.Footer>
    </ModalStyled>
  );
}

export default InstantiationManagementModal;
