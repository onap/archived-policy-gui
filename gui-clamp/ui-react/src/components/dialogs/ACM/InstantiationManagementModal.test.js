/*
 * ============LICENSE_START=======================================================
 *  Copyright (C) 2022 Nordix Foundation.
 *  ================================================================================
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 *  SPDX-License-Identifier: Apache-2.0
 *  ============LICENSE_END=========================================================
 *
 *
 */

import {mount, shallow} from "enzyme";
import React from "react";
import toJson from "enzyme-to-json";
import InstantiationManagementModal from "./InstantiationManagementModal";
import {act} from "react-dom/test-utils";
import {createMemoryHistory} from "history";
import ACMService from "../../../api/ACMService";
import acmLoopList from "./testFiles/acmList.json";
import {BrowserRouter} from "react-router-dom";

const logSpy = jest.spyOn(console, 'log')
const history = createMemoryHistory();

describe('Verify Instantiation Management', () => {
    const flushPromises = () => new Promise(setImmediate);

    beforeEach(() => {
        logSpy.mockClear();
    });

    it("renders without crashing", () => {
        shallow(<InstantiationManagementModal/>);
    });
    it("renders correctly", () => {
        const tree = shallow(<InstantiationManagementModal/>);
        expect(toJson(tree)).toMatchSnapshot();
    });

    it('should have a close Button element', () => {
        const container = shallow(<InstantiationManagementModal/>);
        const button = container.find('[variant="secondary"]').at(2);

        expect(button.text()).toEqual("Close");
    });

    it('should have a Create Instance Button element', () => {
        const container = shallow(<InstantiationManagementModal/>);
        const button = container.find('[variant="primary"]').at(0);

        expect(button.text()).toEqual("Create Instance");
    });

    it('should have a Monitor Instantiations Button element', () => {
        const container = shallow(<InstantiationManagementModal/>);
        const button = container.find('[variant="secondary"]').at(0);

        expect(button.text()).toEqual("Monitor Instantiations");
    });

    it('handleClose called when bottom button clicked', () => {
        const container = shallow(<InstantiationManagementModal history={history}/>);
        const button = container.find('[variant="secondary"]').at(2);

        act(() => {
            button.simulate('click');
            expect(logSpy).toHaveBeenCalledWith('handleClose called');
        });
    });

    it('handleClose called when top-right button clicked', () => {
        const container = shallow(<InstantiationManagementModal history={history}/>);

        act(() => {
            container.find('[size="xl"]').get(0).props.onHide();
            expect(logSpy).toHaveBeenCalledWith('handleClose called');
        });
    });

    it('clearErrors called when clear error message button clicked', () => {
        const container = shallow(<InstantiationManagementModal history={history}/>);
        const button = container.find('[variant="secondary"]').at(1);

        act(() => {
            button.simulate('click');
            expect(logSpy).toHaveBeenCalledWith('clearErrors called');
        });
    });

    it('Check useEffect is being called', async () => {
        jest.resetAllMocks();
        jest.spyOn(ACMService, 'getACMInstantiation')
            .mockImplementationOnce(async () => {
                return Promise.resolve({
                    ok: true,
                    status: 200,
                    text: () => "OK",
                    json: () => {
                        return Promise.resolve(acmLoopList);
                    }
                });
            });

        const component = mount(
            <BrowserRouter>
                <InstantiationManagementModal/>
            </BrowserRouter>
        );
        const useEffect = jest.spyOn(React, "useEffect");

        await act(async () => {
            await flushPromises()
            component.update();
            await expect(useEffect).toHaveBeenCalled();

        });
        component.unmount();
    });

    it('set state gets called for setInstantiationList useEffect on success', async () => {
        const setInstantiationList = jest.fn();
        const setDeleteInstantiation = true;
        const history = createMemoryHistory();
        jest
            .spyOn(global, 'fetch')
            .mockImplementation(() =>
                Promise.resolve({
                    ok: true,
                    status: 200,
                    text: () => "OK",
                    json: () => acmLoopList
                })
            )

        mount(
            <BrowserRouter>
                <InstantiationManagementModal history={history}/>
            </BrowserRouter>
        );
        act(async () => {
            expect(setInstantiationList).toHaveBeenCalledTimes(1);
            expect(setDeleteInstantiation).toHaveBeenCalledTimes(1);
        });
    });
});
