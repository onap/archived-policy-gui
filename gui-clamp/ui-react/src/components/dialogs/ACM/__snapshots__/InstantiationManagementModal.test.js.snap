// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify Instantiation Management renders correctly 1`] = `
<Styled(Modal)
  backdrop="static"
  keyboard={false}
  onHide={[Function]}
  show={true}
  size="xl"
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle>
      Manage Instances
    </ModalTitle>
  </ModalHeader>
  <ModalBody>
    <Container
      fluid={false}
    >
      <Row
        noGutters={false}
      >
        <Link
          to={
            Object {
              "pathname": "/editACMInstanceProperties",
            }
          }
        >
          <Button
            active={false}
            disabled={false}
            type="null"
            variant="primary"
          >
            Create Instance
          </Button>
        </Link>
        <styled.div />
        <Link
          to={
            Object {
              "pathname": "/monitorInstantiation",
            }
          }
        >
          <Button
            active={false}
            disabled={false}
            type="null"
            variant="secondary"
          >
            Monitor Instantiations
          </Button>
        </Link>
      </Row>
    </Container>
    <ForwardRef
      bordered={true}
      style={
        Object {
          "marginTop": "10px",
        }
      }
    >
      <thead>
        <tr>
          <th>
            #
          </th>
          <th
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            Instantiation Name
          </th>
          <th
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            Edit Instantiation
          </th>
          <th
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            Delete Instantiation
          </th>
          <th
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            Change Order State
          </th>
          <th
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            Instantiation Order State
          </th>
          <th
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            Instantiation Current State
          </th>
        </tr>
      </thead>
      <tbody />
    </ForwardRef>
    <styled.div />
  </ModalBody>
  <ModalFooter>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      type="null"
      variant="secondary"
    >
      Clear Error Message
    </Button>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      type="null"
      variant="secondary"
    >
      Close
    </Button>
  </ModalFooter>
</Styled(Modal)>
`;
